Support Linux kernel 5.15.0-116-generic

## Description
When we build Open-cas-linux on a new kernel 5.15.0-116-generic, we get an implicit declaration of the 'module_put_and_exit' function

## Expected Behavior
Compiled package Open-cas-linux in Linux OS  with new kernel 5.15.0-116-generic

## Actual Behavior
Error when compiling a package.

## Steps to Reproduce
Upgrade kernel to 5.15.0-116-generic and compile Open-cas-linux
```
apt upgrade
git clone https://github.com/Open-CAS/open-cas-linux
cd open-cas-linux
git submodule update --init
./configure
make

```
## Logs
root@sw:~/open-cas-linux# make
cd modules && make
make[1]: Entering directory '/root/open-cas-linux/modules'
make[2]: Entering directory '/root/open-cas-linux/ocf'
make[3]: Entering directory '/root/open-cas-linux/ocf'
make[3]: Nothing to be done for 'distcleandir'.
make[3]: Leaving directory '/root/open-cas-linux/ocf'
make[2]: Leaving directory '/root/open-cas-linux/ocf'
make[2]: Entering directory '/root/open-cas-linux/ocf'
make[3]: Entering directory '/root/open-cas-linux/ocf'
make[3]: Nothing to be done for 'distcleandir'.
make[3]: Leaving directory '/root/open-cas-linux/ocf'
make[2]: Leaving directory '/root/open-cas-linux/ocf'
cd "/lib/modules/"5.15.0-116-generic"/build" && make M=/root/open-cas-linux/modules modules
make[2]: Entering directory '/usr/src/linux-headers-5.15.0-116-generic'
  CC [M]  /root/open-cas-linux/modules/cas_cache/layer_cache_management.o
/root/open-cas-linux/modules/cas_cache/layer_cache_management.c: In function ‘exit_instance_finish’:
/root/open-cas-linux/modules/cas_cache/layer_cache_management.c:604:9: error: implicit declaration of function ‘module_put_and_exit’; did you mean ‘complete_and_exit’? [-Werror=implicit-function-declaration]
  604 |         module_put_and_exit(0);
      |         ^~~~~~~~~~~~~~~~~~~
      |         complete_and_exit
/root/open-cas-linux/modules/cas_cache/layer_cache_management.c:605:1: error: control reaches end of non-void function [-Werror=return-type]
  605 | }
      | ^
cc1: all warnings being treated as errors
make[4]: *** [scripts/Makefile.build:297: /root/open-cas-linux/modules/cas_cache/layer_cache_management.o] Error 1
make[3]: *** [scripts/Makefile.build:560: /root/open-cas-linux/modules/cas_cache] Error 2
make[2]: *** [Makefile:1911: /root/open-cas-linux/modules] Error 2
make[2]: Leaving directory '/usr/src/linux-headers-5.15.0-116-generic'
make[1]: *** [Makefile:48: default] Error 2
make[1]: Leaving directory '/root/open-cas-linux/modules'
make: *** [Makefile:22: modules] Error 2


## Your Environment
* OpenCAS version (commit hash or tag):  v22.6.2
* Operating System: Ubuntu 22.04
* Kernel version: 5.15.0-116-generic

--- a/configure.d/1_module_put_and_exit.conf
+++ b/configure.d/1_module_put_and_exit.conf
@@ -1,0 +1,32 @@
+#!/bin/bash
+#
+# Copyright(c) 2012-2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+. $(dirname $3)/conf_framework
+
+
+check() {
+	cur_name=$(basename $2)
+	config_file_path=$1
+    if compile_module $cur_name "module_put_and_kthread_exit(0);" "linux/module.h"
+    then
+        echo $cur_name 1 >> $config_file_path
+    else
+        echo $cur_name 2 >> $config_file_path
+    fi
+}
+
+apply() {
+    case "$1" in
+    "1")
+	    add_define "CAS_MODULE_PUT_AND_EXIT(code) module_put_and_kthread_exit(code)" ;;
+    "2")
+	    add_define "CAS_MODULE_PUT_AND_EXIT(code) module_put_and_exit(code)" ;;
+    *)
+        exit 1
+    esac
+}
+
+conf_run $@

--- a/modules/cas_cache/layer_cache_management.c
+++ b/modules/cas_cache/layer_cache_management.c
@@ -601,7 +601,7 @@ static int exit_instance_finish(void *data)
 	if (result == -KCAS_ERR_WAITING_INTERRUPTED)
 		kfree(ctx);
 
-	module_put_and_exit(0);
+	CAS_MODULE_PUT_AND_EXIT(0);
 }
 
 struct _cache_mngt_attach_context {
@@ -649,7 +649,7 @@ static int cache_start_rollback(void *data)
 	if (result == -KCAS_ERR_WAITING_INTERRUPTED)
 		kfree(ctx);
 
-	module_put_and_exit(0);
+	CAS_MODULE_PUT_AND_EXIT(0);
 
 	return 0;
 }

